instrumentation/spring/spring-kafka-2.7/testing/src/main/java/io/opentelemetry/testing/AbstractSpringKafkaNoReceiveTelemetryTest.java

```diff
@@ -293,19 +321,40 @@ void shouldHandleFailureInKafkaBatchListener() {
                             .hasLinksSatisfying(links(producer.get().getSpanContext()))
                             .hasStatus(StatusData.error())
                             .hasException(new IllegalArgumentException("boom"))
-                            .hasAttributesSatisfyingExactly(
-                                equalTo(SemanticAttributes.MESSAGING_SYSTEM, "kafka"),
-                                equalTo(
-                                    SemanticAttributes.MESSAGING_DESTINATION_NAME,
-                                    "testBatchTopic"),
-                                equalTo(SemanticAttributes.MESSAGING_OPERATION, "process"),
-                                equalTo(
-                                    SemanticAttributes.MESSAGING_KAFKA_CONSUMER_GROUP,
-                                    "testBatchListener"),
-                                satisfies(
-                                    SemanticAttributes.MESSAGING_CLIENT_ID,
-                                    stringAssert -> stringAssert.startsWith("consumer")),
-                                equalTo(SemanticAttributes.MESSAGING_BATCH_MESSAGE_COUNT, 1)),
-                    span -> span.hasName("consumer").hasParent(trace.getSpan(0))));
+                            .hasAttributesSatisfyingExactly(processAttributes),
+                    span -> span.hasName("consumer").hasParent(trace.getSpan(0))),
+            trace -> {
+              if (Boolean.getBoolean("testLatestDeps")) {
+                // in latest dep tests process spans are not created for retries because spring does
+                // not call the success/failure methods on the BatchInterceptor for reties
```

Comment:
is this comment relevant for the javaagent instrumentation?

Reply:
Thanks I left the comment as it is but change the condition so that this branch is only used for library instrumentation. I think ideally we should implement the library instrumentation in a different way to have the library instrumentation create correct traces for retried batches.